core/domain/draft_upgrade_services.py

```diff
@@ -87,6 +87,22 @@ def try_upgrading_draft_to_exp_version(
 class DraftUpgradeUtil(python_utils.OBJECT):
     """Wrapper class that contains util functions to upgrade drafts."""
 
+    @classmethod
+    def _convert_states_v30_dict_to_v31_dict(cls, draft_change_list):
+        """Converts draft change list from state version 30 to 31. State
+        Version 31 adds the duration float for the Voiceover section of state,
+        for which there should be no changes to drafts.
+
+        Args:
+            draft_change_list: list(ExplorationChange). The list of
+                ExplorationChange domain objects to upgrade.
+
+        Returns:
+            list(ExplorationChange). The converted draft_change_list.
+        """
+
+        return draft_change_list
```

Comment:
Suppose the draft_change_list contains the changes which adds audio, so do we have to update the draft change list in that case?

Reply:
I'm going to test this out and give some feedback.

Reply:
From what I understand of the Exploration and ExplorationChange class, it looks like that this draft_upgrade_service uses the command system to make changes to data to the exploration or the state. What would be the difference with using commands as opposed to the _convert_states_vX_dict_to_vY_dict method?

Reply:
Resolved by doing the proper draft_changes update. So done.